### ISSUE TITLE
Feature: find module symbols by name, with an Open dialog

### ISSUE THREAD

**Issue by pvginkel** [RustDT/RustDT#40]
The `Ctrl+Shift+R` shortcut in eclipse allows you to find files. However, with Rust, a lot of files are named `mod.rs`. I'm not sure whether this is possible, but it would be very useful if these were differentiated in some way. E.g. if a file is named `mod1/mod.rs`, that typing `mod1` would also match that file.

**Comment by Koka** [comment]
Sounds like something that's already implemented in eclipse [itself](http://stackoverflow.com/questions/5752342/when-typing-in-open-resource-dialog-is-there-a-way-to-find-files-by-directory-i).

**Comment by bruno-medeiros** [comment]
True, purely for files, you can use Open Resource (Ctrl-Shift-R). 

What I would like to do though, is implement a functionality similar to JDT's "Open Type" (Ctrl-Shift-T), where you can search for type symbols (in the case of Java, class or interface types).

**Comment by Woyten** [comment]
@bruno-medeiros I am currently working on the "Open Type" feature. My idea was to build up an index for the workspace using `parse-describe`and then show a popup similar to the "Quick Outline".

Please let me know if some work on that feature is already ongoing or if there are major infrastructure changes on your stash that are expected to conflict with my development.

**Comment by bruno-medeiros** [comment]
@Woyten  Nope, there isn't anything planned for the near future that would interfere with that. The major features I have my eyes is Language Server Protocol integration (https://code.visualstudio.com/blogs), and possibly even a port of RustDT/MelnormeEclipse to IntelliJ IDEA (!). 

BTW, if you have more development questions, you might want to join the RustDT Gitter: https://gitter.im/RustDT/RustDT
